(Backport 51074) disk: support setting FAT size for format_

aplanas opened the pull request:
### What does this PR do?

Add new optional parameter `fat` in disk.format_ to specify the FAT size (12, 16 or 32).

### Tests written?

Yes

(backport #51074, already merged in develop)

mirceaulinic committed:
Allow the __utils__ dunder in Grains modules

mattp- committed:
Revert "Revert "use the minion configured publish_port, not what is provided in auth payload""

This restores original commit a0d723f78f3f6196e11f70058dd9056fbed16676,
alongside some changes in tests.integration syndic config generation
that was actually subtly doing the wrong thing and copying from a base
of the base master, not syndic master config.

Conflicts:
  - tests/integration/__init__.py

aplanas committed:
disk: support setting FAT size for format_

(cherry picked from commit 5828aa1730022ce7d46a53507dade94dde44fa67)

waynew committed:
Update network states doc

Previously it was xmit_hash_policy, but that's not what the code is
actually looking for.

cbosdo committed:
Add virt.volume_infos()

Expose informations of the libvirt storage volumes, either for one or
all the volumes. Among the provided data, this function exposes the
names of the virtual machines using the volumes of file type.

cbosdo committed:
Add virt.volume_delete function

All users to remove a libvirt storage volume.

cbosdo committed:
virt.volume_info: handle disappeared volumes

It may happen on pools that have not refresh for a long time that the
volume list is not matching reality. Then when calling the info() or
XMLDesc() functions libvirt errors out.

virt.volume_info() now handles this case.

cbosdo committed:
virt.volume_infos parameters now default to None

Since the pool and volume parameters can be None, make them optional and
default to None.

Tanoti committed:
Add support for new config option 'git_pillar_interval' so git_pillar data can be collected less frequently than the main loop_interval timer allows

cbosdo committed:
virt.volume_infos fix for single VM

_get_domain returns a domain object when only one VM has been found.
virt.volume_infos needs to take care of it or it will fail to list the
volumes informations if the host in such a case.

twangboy committed:
Merge pull request #52293 from afischer-opentext-com/slsutil_renderer_support_saltenv

Looks great! Thanks for the code!

github-abcde committed:
Added add/delete/list/exists upstream functions.
Pylint-inspired layout changes.
Also remove empty lines in _safe_output function.

github-abcde committed:
Add state for managing RabbitMQ upstreams.

github-abcde committed:
Added version-added docstrings. Added parameter documentation.

github-abcde committed:
Added version-added docstrings. Updated parameter documentation layout. Renamed dicttrim.remove_empty_values to data.filter_falsey as per PR 52499.

github-abcde committed:
salt/modules/rabbitmq.py: Changed add_upstream from using one "definition"-parameter to using all the parameters that make up the definition.

github-abcde committed:
Rename add_upstream to set_upstream as that fits better.
Refactored upstream_present to remove duplicate code.

github-abcde committed:
tests/unit/modules/test_rabbitmq.py: Added tests for list_upstream, set_upstream, delete_upstream. Pylint-inspired fixes.

github-abcde committed:
Replaced incorrect doc-directive "version-added" with the correct directive.

github-abcde committed:
salt/modules/rabbitmq.py: Docstring fixes.

github-abcde committed:
salt/modules/rabbitmq.py: More docstring fixes.

max-arnold committed:
Add tests to check for slot expansion in unless/onlyif

max-arnold committed:
Parse slots in unless/onlyif module arguments

github-abcde committed:
Doc updates.
Fixed docstring for salt/states/rabbitmq_upstream.py:present

sathieu committed:
Rename git_pillar_interval to git_pillar_update_interval

Similar to gifs_update_interval

sathieu committed:
Add doc and salt.config code for git_pillar_update_interval

max-arnold committed:
Document slot parsing improvements in Neon release notes

https://github.com/saltstack/salt/pull/52351
https://github.com/saltstack/salt/pull/53307

Oloremo committed:
Added ES index get_settings and put_settings methods

github-abcde committed:
Added files for boto3_elasticsearch:
utils/boto3_elasticsearch.py: Contains custom waiter definitions that we will need to use until they get implemented in botocore (not present in 1.12.183).
modules/boto3_elasticsearch.py: Contains all functions from boto3 that interact with AWS. Added a few helper functions.
states/boto3_elasticsearch.py: Added states for the AWS ES service.

github-abcde committed:
Added unittests for boto3_elasticsearch.

github-abcde committed:
salt/modules/boto3_elasticsearch.py: Removed debugcode.

github-abcde committed:
Added yaml code examples in docstrings.

github-abcde committed:
Removed ESUpgradeFinished waiter definition as this was not accurate enough.
Renamed ESUpgradeFinished2 to ESUpgradeFinished instead.
Renamed references to ESUpgradeFInished2 in the execution module.

github-abcde committed:
Fixed tests with the changed structure of errormessages.

github-abcde committed:
salt/states/boto3_elasticsearch.py: Fix pylint warnings.

github-abcde committed:
Added and fixed documentation.

github-abcde committed:
salt/modules/boto3_elasticsearch.py: Removed trivial CLI examples. Added nontrivial CLI examples. Fixed broken CLI example.

github-abcde committed:
salt/states/boto3_elasticsearch.py: Deduplicated duplicate code into _check_return_value

github-abcde committed:
salt/modules/boto3_elasticsearch.py: Fix incorrect CLI example.

cbosdo committed:
virt.volume_infos: don't raise an error if there is no VM

virt._get_domain() throws an CommandExecutionError if there is no
virtual machine. In volume_infos() this is not an error at all since the
VMs list is only used to find out which VM uses each volume.

cbosdo committed:
virt.volume_infos: silence libvirt error message

Even though the volume_infos handles the libvirt exception when a volume
is missing, libvirt was still outputting the error message in the log.
Since this can add noise to the log only record the libvirt error
message in debug level.

max-arnold committed:
Add optional warnings to test.configurable_test_state

max-arnold committed:
Fix default argument docstring

sbrennan4 committed:
Set session_id cookie

sbrennan4 committed:
Fix test

cbosdo committed:
virt.network_define allow adding IP configuration

If using virt.network_define with nat network type, then libvirt
complains about missing IP configuration. Allow setting it in both the
virt.network_define module and the virt.network_running state.

cbosdo committed:
virt.volume_infos needs to ignore inactive pools

libvirt raises an error when getting the list of volumes of a pool that
is not active. Rule out those pools from virt.volume_infos since we
still need to give infos on the other pools' volumes.

sathieu committed:
Fix ldap.managed errors

Regression from 05e55bd (#48666):
- AttributeError: 'tuple' object has no attribute 'encode'
- ValueError: too many values to unpack

Fixes: #52022.

cbosdo committed:
Fix virt._gen_pool_xml when all source parameters are None

When all the source_* parameters of _gen_pool_xml are None, no <source>
element should be generated. Instead we had the following:

    <source><path dir="None"/></source>

Rather than filling a source structure with empty members, directly pass
a None for the source as expected by the jinja template.

cbosdo committed:
Add virt.pool_update function

Add a function that allows changing a virtual storage pool. By using the
test=True parameter, this function can be used to check if a change
would be done.

cbosdo committed:
virt.pool_running state improvements

With this commit the virt.pool_running state is capable to update an
existing pool before ensuring it is running.

This also adds support for the test parameter.

cbosdo committed:
Fix virt.pool_define documentation

virt.pool_define example are mentioning a uuid property while this one
should be named value. The code was right, but not the doc.

Commit:
virt: adding kernel boot parameters to libvirt xml

SUSE's autoyast and Red Hat's kickstart take advantage of kernel paths,
initrd paths, and kernel boot command line parameters. These changes
provide the option of using these, and will allow salt and
autoyast/kickstart to work together.

Signed-off-by: Larry Dewey <ldewey@suse.com>

cbosdo committed:
Add virt.pool_capabilities function

Not all storage backends are supported for a given libvirt hypervisor. For the user
to know what is supported and what is not expose the recently added
libvirt function providing pool capabilities.

For older libvirt version, let's craft data that are reasonable by
adding a computed flag to warn these data may not be 100% accurate.

silenius committed:
fix ##54448

Commit:
Docs: Make cloud.create example readable

Commit:
Fix up the stalekey engine docs

cbosdo committed:
Add source_initiator parameter in virt.pool_define

In order to define iscsi-direct virtual storage pools, the use needs to
be able to provide the initiator IQN. Add a parameter for it in:
  * virt.pool_define module function
  * virt.pool_update module function
  * virt.pool_running state

cbosdo committed:
virt pool secret have no type attribute

According to libvirt schemas and doc only the volume secrets have a
type.

cbosdo committed:
virt: create and update pool secret's password for the user

Libvirt stores RBD and iSCSI passwords in secrets. Add a password field
in the source_auth parameter of virt.pool_define and virt.pool_update to
set the password value rather than reuse an already defined libvirt
secret.

cbosdo committed:
virt.pool_define: remove potential leading / in CIFS source path

libvirt needs the CIFS source path not to start with a /. Let's remove
them since this could be a common user mistake.

Commit:
virt: Download linux and initrd

Signed-off-by: Larry Dewey <ldewey@suse.com>

max-arnold committed:
Change versionadded to Neon for test state warnings arg

Ch3LL committed:
Add deprecations to changelog and neon release notes

Ch3LL committed:
do not deprecate rand-str

Ch3LL committed:
include deprecation notice for firewalld

dwoz reviewed (changes requested):
@aplanas Looks like the mac os test failures are related to this change.

dwoz committed:
Merge branch 'master' into 2019_2_1_port_49076

dwoz reviewed (changes requested):
@aplanas The mac test failures seem too be related.

dwoz committed:
Merge pull request #54992 from mchugh19/port-51846

master-port 51846 and 52969

dwoz committed:
Merge pull request #55718 from dwoz/flaky_win_test

Reduce flakiness in test on windows

dwoz committed:
Merge pull request #55684 from Ch3LL/bp_50344

Master Port #50344

dwoz committed:
Merge pull request #55643 from Ch3LL/issue_49418

Ensure refresh and refresh_db both work in aptpkg

dwoz committed:
Merge pull request #55614 from Ch3LL/issue_49428

Deprecate jinja filters for neon

dwoz committed:
Merge pull request #55679 from Ch3LL/issue_49413

document retry_on_errors and error_retries boto opts and remove deprecation

dwoz committed:
Merge pull request #55683 from Ch3LL/bp_50343

Master Port #50343

dwoz committed:
Merge pull request #55685 from Ch3LL/bp_49826

Master Port #49826

dwoz committed:
Merge pull request #55682 from Ch3LL/bp_49796

Master Port #49796

dwoz committed:
Merge pull request #55641 from Ch3LL/issue_49417

Deprecate enviroment for environment in heat modules

dwoz committed:
Merge pull request #55569 from Ch3LL/issue_49423

 Deprecate nova cloud driver

dwoz committed:
Merge pull request #55573 from Ch3LL/issue_49422

Deprecate quiet kwarg in cmd.run state module

dwoz committed:
Merge pull request #55568 from Ch3LL/issue_49424

Document use of ? to split path/key in vault sdb is deprecated

dwoz committed:
Merge pull request #55217 from MichaelSaah/patch-1

Note known pip version incompatibility

Ajnbro committed:
Merge branch 'master' into http_query_update

Ch3LL committed:
Deprecate salt/utils/__init__.py functions

Ch3LL committed:
add utils deprecations to chagnelog/neon release notes

Ch3LL committed:
fix neon docs

Ch3LL committed:
Import correct six module

Ch3LL committed:
fix utils for tests

Ch3LL committed:
Deprecate gitfs_env_whitelist and gitfs_env_blacklist

Ch3LL committed:
remove gitfs_env_blacklist from tests

Ch3LL committed:
add gitfs fileserver deprecation notice

Ch3LL committed:
Bump deprecation of override_name to sodium

cdalvaro committed:
Merge branch 'master' into improve_macOS_homebrew_cask_support

cdalvaro committed:
Merge branch 'master' into 2019.2.1

dwoz committed:
Merge pull request #55716 from s0undt3ch/features/pytest-staged-31

Address flakiness of test_salt_documentation

dwoz committed:
Merge pull request #55706 from dwoz/zabbix-exceptiono

Update zabbix module to avoid exception

dwoz committed:
Merge pull request #55715 from Ch3LL/changelog_neon

Add deprecations to changelog and neon release notes

github-abcde committed:
Merge branch 'master' into depends_decorator-version-master

waynew committed:
Merge branch 'master' into configurable-test-state-warning

Ch3LL committed:
add gitfs_saltenv_whiltelist/blacklist tests

Ch3LL committed:
test correct deprecated env opts

Ch3LL committed:
fix utils path and is_windows imports for tests

dwoz committed:
Merge pull request #54916 from Oloremo/added_http_request_interval

Added request_interval feature to wait_for_successful_query module

dwoz committed:
Merge pull request #55721 from Oloremo/53736-added-es-index-settings

Added ES index get_settings and put_settings methods

dwoz committed:
Merge pull request #55723 from max-arnold/renderer-saltenv

[master] Port #52293 (saltenv support in slsutil.renderer)

dwoz committed:
Merge pull request #55700 from Oloremo/55443-archive-hash-trust

Added a skip_files_list_verify argument to archive.extracted state

dwoz committed:
Merge pull request #54665 from garethgreenaway/2019_2_1_port_52710

[master] Porting #52710 to master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52747

dwoz committed:
Merge pull request #55202 from cbosdo/master-pool-running-improvements

pool running improvements

dwoz committed:
Merge branch 'master' into http_query_update

dwoz committed:
Merge pull request #54686 from garethgreenaway/2019_2_1_port_51098

[master] Porting #51098 to master

dwoz committed:
Merge branch 'master' into master-get-full-returns-jid-fix

dwoz committed:
Merge pull request #54650 from garethgreenaway/2019_2_1_port_51538

[master] Porting #51538 to master

dwoz committed:
Merge pull request #55726 from Ch3LL/dep_context

Bump deprecation of override_name to sodium

Ch3LL committed:
Add publish_port transport tcp/zeromq tests

Ch3LL committed:
use salt.utils.versions.warn_until instead of __utils__

dwoz committed:
Merge pull request #53621 from sathieu/git_pillar_update_interval

Git pillar update interval

dwoz committed:
Fix linter

cdalvaro committed:
Merge branch 'master' into 2019.2.1

github-abcde committed:
Port changes from PR 52499 to master

dwoz committed:
Merge pull request #54675 from garethgreenaway/2019_2_1_port_49076

[master] Porting #49076 to master

dwoz committed:
Merge pull request #54678 from garethgreenaway/2019_2_1_port_52747

[master] Porting #52747 to master

dwoz committed:
Merge pull request #53959 from max-arnold/configurable-test-state-warning

Add optional warnings to test.configurable_test_state

dwoz committed:
linter again

dwoz committed:
Merge pull request #54960 from aplanas/backport_50452

(Backport 50452, 54022) Add new module freezer

dwoz committed:
Merge pull request #54961 from aplanas/backport_50541

(Backport 50541) btrfs: add all subvolume commands

dwoz committed:
Merge pull request #55510 from FerrySchuller/patch-4

num_cpus and cpu_model support for IBM/S390

dwoz committed:
Merge pull request #55651 from sathieu/fix-ldap-managed

Fix ldap.managed errors

dwoz committed:
Merge pull request #54216 from cdalvaro/improve_macOS_homebrew_cask_support

Improve macOS homebrew cask support

dwoz committed:
Merge pull request #55722 from Ch3LL/dep_utils

Deprecate salt/utils/__init__.py functions

dwoz committed:
Fix linter

dwoz committed:
Merge pull request #55730 from Ch3LL/mp_50089

Master Port #50089

cdalvaro committed:
bugfix: Solve pylint old-style class warning

cdalvaro committed:
Merge branch '2019.2.1' of github.com:cdalvaro/salt into 2019.2.1

dwoz committed:
Merge pull request #55725 from Ch3LL/dep_gitfs

Deprecate gitfs_env_whitelist and gitfs_env_blacklist

dwoz committed:
Merge pull request #55664 from Ch3LL/issue_49430

Bump deprecation warning to Aluminium for neutron module

s0undt3ch committed:
Use coverage==5.0.1

asomers committed:
Fix the mem_total and swap_total tests on FreeBSD and OpenBSD

They were comparing a float to an int.  AFAICT these tests have never
passed.  Tested on FreeBSD 13 on the master branch.

dwoz committed:
Merge pull request #55734 from dwoz/lint_fix

Fix linter

asomers committed:
Fix the test_patch*failure tests on FreeBSD

FreeBSD's patch command has a slightly different error message than GNU
patch.  Adjust the tests' expectations to accept either error message.

Tested on FreeBSD 13 with the master branch.

waynew committed:
Merge branch 'master' into 2019_2_1_port_52365

dwoz committed:
Merge pull request #55733 from ogd-software/data_filter_falsey-master

Data filter falsey-master

dwoz committed:
Merge branch 'master' into master-pool-capabilities

dwoz committed:
Merge pull request #55145 from Ajnbro/http_query_update

Added Status Code Lists for HTTP Query State

dwoz committed:
Merge branch 'master' into master-get-full-returns-jid-fix

dwoz committed:
Merge branch 'master' into master-pr52033

dwoz committed:
Merge branch 'master' into fix_humanname

dwoz committed:
Merge pull request #55218 from nembery/patch-1

update openstack.py for more verbose auth example

dwoz committed:
Merge pull request #55155 from multani/fix-doc-execution-s3

Fix the documentation rendering for the S3 execution module

dwoz committed:
Merge pull request #54769 from cdalvaro/2019.2.1

Call bash only when necessary on macOS

dwoz committed:
Merge pull request #54660 from garethgreenaway/2019_2_1_port_52244

[master] Porting #52244 to master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_53160

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52124

dwoz committed:
Merge pull request #55045 from mattLLVW/fix_mysql_returner_schema

fix mysql returner schema

asomers committed:
Microoptimize the command to set FreeBSD's virtual grain

dwoz committed:
Merge pull request #54657 from garethgreenaway/2019_2_1_port_49250

[master] Porting #49250 to master

dwoz committed:
Merge pull request #54982 from mchugh19/port-50215

master-port 50215 and 51943

dwoz committed:
Merge pull request #54662 from garethgreenaway/2019_2_1_port_50399

[master] Porting #50399 to master

dwoz committed:
Merge pull request #54601 from garethgreenaway/2019_2_1_port_53160

[master] Porting #53160 to master

dwoz committed:
Merge branch 'master' into sysctl_n

github-abcde committed:
Merge branch 'master' into depends_decorator-version-master

s0undt3ch committed:
Bump PyTest Salt package

Previous pytest-salt packages didn't include the `sitecustomize.py` file
required for coverage tracking of multiprocessing processes.

This new version fixes that.

asomers committed:
Skip test_lookup_with_servers if nslookup is not installed

waynew committed:
Add note to changelog

waynew committed:
Merge branch 'master' into fix-wusa-3010-error

s0undt3ch committed:
Don't forget to format the string

waynew committed:
Merge branch 'master' into 2019_2_1_port_52420

garethgreenaway committed:
Fixing string declarations.

dwoz committed:
Merge pull request #54641 from garethgreenaway/2019_2_1_port_50485

[master] Porting #50485 to master

dwoz committed:
Merge pull request #55346 from cbosdo/master-pool-capabilities

Add virt.pool_capabilities function

dwoz committed:
Merge branch 'master' into master-pr52033

dwoz committed:
Merge pull request #54590 from garethgreenaway/2019_2_1_port_52124

[master] Porting #52124 to master

dwoz committed:
Merge pull request #55590 from ogd-software/depends_decorator-version-master

Depends decorator version master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_49844

dwoz committed:
Merge branch 'master' into 2019_2_1_port_50842

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52609

dwoz committed:
Merge branch 'master' into 2019_2_1_port_51168

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52150

dwoz committed:
Merge branch 'master' into 2019_2_1_port_51929

dwoz committed:
Merge branch 'master' into 2019_2_1_port_51954

dwoz committed:
Merge branch 'master' into 2019_2_1_port_50117

dwoz committed:
Merge branch 'master' into 2019_2_1_port_50938

asomers committed:
Fix integration.modules.test_sysctl.SysctlModuleTest.test_show_freebsd

This test was originally committed incorrectly, but it passed by
accident on Python 2.  However, Python 3 does not allow comparisons
between lists and ints, so now the test fails.

asomers committed:
Skip DiskModuleVirtualizationTest.test_no_mtab on FreeBSD

FreeBSD does not use mtab.

dwoz committed:
Merge pull request #54591 from garethgreenaway/2019_2_1_port_52609

[master] Porting #52609 to master

dwoz committed:
Merge pull request #54600 from garethgreenaway/2019_2_1_port_51168

[master] Porting #51168 to master

dwoz committed:
Fix missing variable wart

dwoz committed:
Merge pull request #55738 from asomers/misc_test_fixes

Misc test fixes

dwoz committed:
fix error when exception message is not ascii

dwoz committed:
Merge pull request #53307 from max-arnold/slot-args

Fix slot parsing in unless/onlyif function arguments

asomers committed:
Fix StatusModuleTest.test_status_pid on FreeBSD

`ps aux` was insufficient on FreeBSD because the default output width
isn't wide enough to capture all of python's arguments (like "salt").
Using `pgrep -f` is more predictable and cross-platform.

asomers committed:
Implement PkgModuleTest.test_pkg_latest_version for FreeBSD

asomers committed:
Add an integration test for pkg.which

dwoz committed:
Merge pull request #53462 from lorengordon/remote-sources

Supports all valid protos for remote sources

dwoz committed:
Merge pull request #55743 from waynew/bloomberg-tornado-set-cookie

Bloomberg tornado set cookie

dwoz committed:
Merge pull request #54411 from axa-group/fix-wusa-3010-error

Manage wusa 3010 return code

dwoz committed:
Merge pull request #55739 from asomers/sysctl_n

Microoptimize the command to set FreeBSD's virtual grain

dwoz committed:
Merge pull request #55737 from asomers/test_patch_regex

Fix the test_patch*failure tests on FreeBSD

s0undt3ch committed:
Fix `integration.cloud.clouds.test_vultrpy.VultrTest.test_list_images`

max-arnold committed:
Port json_query Jinja filter from Ansible

dwoz committed:
Merge pull request #55744 from s0undt3ch/features/pytest-master-3

Don't forget to format the string

dwoz committed:
Merge pull request #55748 from saltstack/features/pytest-master-4

Fix `integration.cloud.clouds.test_vultrpy.VultrTest.test_list_images`

dwoz committed:
Merge pull request #55216 from brejoc/master-get-full-returns-jid-fix

Checking for jid before returning data

dwoz committed:
Merge pull request #55165 from cbosdo/master-pr52033

Master -- pr52033 -- Volume infos fix

dwoz committed:
Merge pull request #54642 from garethgreenaway/2019_2_1_port_50938

[master] Porting #50938 to master

dwoz committed:
Merge pull request #54602 from garethgreenaway/2019_2_1_port_52150

[master] Porting #52150 to master

dwoz committed:
Merge pull request #54580 from garethgreenaway/2019_2_1_port_50842

[master] Porting #50842 to master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52365

dwoz committed:
Merge pull request #54569 from garethgreenaway/2019_2_1_port_52365

[master] Porting #52365 to master

dwoz committed:
Merge pull request #55736 from asomers/bsd_memtotal_test_2018.3

Fix the mem_total and swap_total tests on FreeBSD and OpenBSD

dwoz committed:
Merge pull request #55753 from asomers/misc_test_fixes2

Fix several tests that fail on FreeBSD

dwoz committed:
Merge pull request #55749 from max-arnold/json-query-port

[master] Port json_query Jinja filter from Ansible

dwoz committed:
Merge pull request #54630 from garethgreenaway/2019_2_1_port_50117

[master] Porting #50117 to master

s0undt3ch committed:
Run `pre-commit run -av pip-tools-compile` against the changes from #55749

This change doesn't really update any requirements files, just it's
comments because `jmespath` is now a primary dependency, not a
secondary.

github-abcde committed:
Remove NO_MOCK and NO_MOCK_REASON

dwoz committed:
Merge pull request #55741 from s0undt3ch/features/pytest-master-2

Bump PyTest Salt package

dwoz committed:
Merge pull request #54583 from garethgreenaway/2019_2_1_port_49844

[master] Porting #49844 to master

dwoz committed:
Merge pull request #54611 from garethgreenaway/2019_2_1_port_51929

[master] Porting #51929 to master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_51954

dwoz committed:
Merge pull request #55245 from ldeweysuse/kernel_params

virt: adding kernel boot parameters to libvirt xml

dwoz committed:
Merge pull request #55515 from aplanas/backport_51704

(Backport 51704) parted: fix set_ valid flags comment

dwoz committed:
Merge branch 'master' into 2019_2_1_port_50083

dwoz committed:
Merge pull request #54629 from garethgreenaway/2019_2_1_port_51053

[master] Porting #51053 to master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_50023

dwoz committed:
Merge branch 'master' into 2019_2_1_port_52420

dwoz committed:
Merge pull request #54593 from garethgreenaway/2019_2_1_port_52420

[master] Porting #52420 to master

dwoz committed:
Merge pull request #55763 from s0undt3ch/hotfix/pr-submissions

Run `pre-commit run -av pip-tools-compile` against the changes from #55749

dwoz committed:
Merge branch 'master' into 2019_2_1_port_51954

dwoz committed:
Merge pull request #54603 from garethgreenaway/2019_2_1_port_51954

[master] Porting #51954 to master

dwoz committed:
Merge branch 'master' into 2019_2_1_port_50706

dwoz committed:
Merge pull request #54627 from garethgreenaway/2019_2_1_port_51692

[master] Porting #51692 to master

dwoz committed:
Merge pull request #54577 from garethgreenaway/2019_2_1_port_52715

[master] Porting #52715 to master

dwoz committed:
Merge pull request #54613 from garethgreenaway/2019_2_1_port_51074

[master] Porting #51074 to master

dwoz committed:
Merge pull request #54632 from garethgreenaway/2019_2_1_port_50083

[master] Porting #50083 to master

dwoz committed:
Merge pull request #55050 from Akm0d/fix_novaclient_api

Fix novaclient api

dwoz committed:
Merge pull request #55280 from s0undt3ch/hotfix/coverage-5.0

Switch the new coverage 5.x library

dwoz committed:
Merge pull request #55336 from max-arnold/utils-in-grains

[master] Allow the __utils__ dunder in Grains modules (port #49128)

dwoz committed:
Merge pull request #55516 from aplanas/backport_52962

(Backport 52962) mount: add not_change parameter to set_fstab and similars

dwoz committed:
Merge pull request #55389 from Nate-Watts/port-50579

Port 50579

dwoz committed:
Merge pull request #55225 from Poulpatine/fix_humanname

Remove doc references to humanname in apt/PPA repo

dwoz committed:
Merge pull request #50865 from Denislb/fix_unicode_error

fix error when exception message is not ascii

dwoz committed:
Merge pull request #55767 from ogd-software/rabbitmq_additions-master

Rabbitmq additions master

dwoz committed:
Merge pull request #55768 from ogd-software/boto3_elasticsearch-master

Boto3 elasticsearch master

dwoz committed:
Merge pull request #54197 from cbosdo/virt-net-nat

Extend `virt.network_define` to create NAT networks

dwoz committed:
Merge pull request #55374 from silenius/issue_54448

fix #54448

dwoz committed:
Merge pull request #54620 from garethgreenaway/2019_2_1_port_50023

[master] Porting #50023 to master

dwoz committed:
Merge pull request #55776 from oeuftete/misc-doc-fixes-2019-12

Doc fixes: stalekey engine; cloud.create example

dwoz committed:
Merge pull request #54595 from garethgreenaway/2019_2_1_port_51344

[master] Porting #51344 to master

dwoz committed:
Merge pull request #54604 from garethgreenaway/2019_2_1_port_50706

[master] Porting #50706 to master

garethgreenaway committed:
Skip tests if mkfs binary is not available

garethgreenaway committed:
Merge branch 'backport_51074' of github.com:aplanas/salt into backport_51074
